### ISSUE TITLE
Bug: (sort-classes) Warning for overloads

### ISSUE THREAD

**Issue by FStefanni** [azat-io/eslint-plugin-perfectionist#734]
### Describe the bug

Hi,

I get this warning:

```text
Extra spacing between "importPemKey" and "importPemKey".eslint[perfectionist/sort-classes](https://perfectionist.dev/rules/sort-classes)
```

With this config:

```js
"perfectionist/sort-classes": [
            "warn",
            {
                "groups": [
                    {"group": "public-static-property", "newlinesInside": 0},
                    {"group": "public-property", "newlinesInside": 0},
                    {"group": "protected-static-property", "newlinesInside": 0},
                    {"group": "protected-property", "newlinesInside": 0},
                    {"group": "private-static-property", "newlinesInside": 0},
                    {"group": "private-property", "newlinesInside": 0},

                    "constructor",

                    // --- PUBLIC ---
                    "public-static-get-method",
                    "public-static-set-method",
                    "public-static-method",
                    "public-get-method",
                    "public-set-method",
                    "public-method",
                    "public-abstract-get-method",
                    "public-abstract-set-method",
                    "public-abstract-method",

                    // --- PROTECTED ---
                    "protected-static-get-method",
                    "protected-static-set-method",
                    "protected-static-method",
                    "protected-get-method",
                    "protected-set-method",
                    "protected-method",
                    "protected-abstract-get-method",
                    "protected-abstract-set-method",
                    "protected-abstract-method",

                    // --- PRIVATE ---
                    "private-static-get-method",
                    "private-static-set-method",
                    "private-static-method",
                    "private-get-method",
                    "private-set-method",
                    "private-method",

                    "unknown"
                ],
                "ignoreCase": true,
                "newlinesBetween": 1,
                "newlinesInside": 1,
                "order": "asc",
                "type": "alphabetical"
            }
        ],
```

Where I was not expecting a warning: I like the empty line between the overloads.

Just as a side note, if I remove the empty lines between the overloads, that warning goes away, but a new one comes up from a different plugin that I use (stylistic):

```text
Expected blank line between class members.eslint[@stylistic/lines-between-class-members](https://eslint.style/rules/lines-between-class-members)
```

Rule config:

```js
"@stylistic/lines-between-class-members": [
            "warn",
            "always",
            {
                "exceptAfterOverload": false,
                "exceptAfterSingleLine": true
            }
        ],
```

But I like the second warning. It is just sort-classes that somehow "fails" at recognizing the overloads.

Regards

### Code example

```ts
export abstract class Sign
{
    public static async importPemKey(
        pem: string,
        type: "public"
    ): Promise<crypto.webcrypto.CryptoKey>

    public static async importPemKey(
        pem: SecretString,
        type: "private"
    ): Promise<crypto.webcrypto.CryptoKey>

    public static async importPemKey(
        pem: string | SecretString,
        type: "private" | "public"
    ): Promise<crypto.webcrypto.CryptoKey>
    {
          ...
    }
}
```

### ESLint version

v9.39.4

### ESLint Plugin Perfectionist version

5.8.0

### Validations

- [x] Read the [docs](https://perfectionist.dev).
- [x] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hi @FStefanni 

> But I like the second warning. It is just sort-classes that somehow "fails" at recognizing the overloads.

From what I understand, perfectionist does recognize overloads correctly, the current behavior is to always enforce no newline between them.

It was made like that because it's the generally followed convention (`@stylistic/lines-between-class-members` ignores checking overloads by default), but I understand it can feel restrictive.

We could go with a very simple `newlinesBetweenOverload?: number` option, by default `0` to keep the current behavior isofunctional, and similar to `newlinesBetween`. It's a bit of a "weak" option IMO as I don't think it would likely be overriden though.

**Comment by FStefanni** [comment]
Hi,

thank you for the answer.
Yes, it seems to me that the option you propose should cover my specific case, but it is more generic to cover all possible cases, and this is fine.

Regards

**Comment by azat-io** [comment]
Added in a4b425dba8129b64fe2b50f0586b3b506e77da17.
Released in [v5.9.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v5.9.0).
